### PR TITLE
Add layers of security in comment APIs

### LINKED ISSUE #1064: Add useGuard Check-Abilities on all Comments API endpoints
## Background Information
Currently, there is no API security-layer to prevent unauthorized users to comment

## Steps to Reproduce
Send a POST request to the comment endpoint as an regular user

## How
We've already implemented the Check-abilities guard, we just need to include those decorators on this endpoint

Expected Behavior
- Only authorized users should have the permission to add comments

### PR DESCRIPTION
**Pull request by LuizFNJ**
# Description
*The objective of this PR is to create a layer of security for each comments api by changing permissions so that regular users cannot create, update or delete comments.*

Fixes #1064  (issue)

# Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)     
- [ ] New feature (non-breaking change which adds functionality)     
- [X] Existing feature enhancement (non-breaking change which modifies existing functionality)     

# Testing
*Used Insomnia to test backend comment APIs.*
**With RegularUser:**
Captura de tela de 2024-10-30 17-15-25
*Returns access denied*  

**With ReviewerUser:**
Captura de tela de 2024-10-30 17-22-36
*Returns found user and user Id*  

# Developer Checklist
## General
- [ ] Code is appropriately commented, particularly in hard-to-understand areas     
- [ ] Repository documentation has been updated (Readme.md) with additional steps required for a local environment setup.
- [x] No `console.log` or related logging is added.
- [x] No code is repeated/duplicated in violation of DRY. The exception to this is for new (MVP/Prototype) functionality where the abstraction layer may not be clear (comments should be added to explain the violation of DRY in these scenarios).   
- [ ] Documented with TSDoc all library and controller new functions

## Frontend Changes
- [ ] No new styling is added through CSS files (Unless it's a bugfix/hotfix)
- [ ] All types are added correctly

## Backend Changes
- [ ] All endpoints are appropriately secured with Middleware authentication
- [x] All new endpoints have a interface schema defined

### Tests
- [ ] All existing unit and end to end tests pass across all services     
- [ ] Unit and end to end tests have been added to ensure backend APIs behave as expected     

# Merge Request Review Checklist 
- [ ] An issue is linked to this PR and these changes meet the requirements outlined in the linked issue(s)     
- [ ] High risk and core workflows have been tested and verified in a local environment.     
- [ ] Enhancements or opportunities to improve performance, stability, security or code readability have been noted and documented in JIRA issues if not being addressed.     
- [ ] Any dependent changes have been merged and published in downstream modules     
- [ ] Changes to multiple services can be deployed in parallel and independently. If not, changes should be broken out into separate merge requests and deployed in order.

### CODE REVIEW

File: server/review-task/comment/comment.module.ts [hunk 1]
```diff
@@ -13,7 +15,7 @@ export const CommentModel = MongooseModule.forFeature([
 ]);
 
 @Module({
-    imports: [CommentModel, UsersModule],
+    imports: [CommentModel, UsersModule, ConfigModule, AbilityModule],
```
Comment (caneppelevitor) [inline server/review-task/comment/comment.module.ts:18]: question: why the import of `ConfigModule` ?

Comment (LuizFNJ) [reply server/review-task/comment/comment.module.ts:18]: From what I understand when doing this task, the configModule is used to manage environment variables, in this case for users' tokens when logged in. I found some examples on aletheia of this security layer using the same, correct me if I'm wrong. As I said, I'm having difficulty testing using insomnia

**Review by caneppelevitor — APPROVED** [review]

**Review by pepermao — CHANGES_REQUESTED** [review]

File: server/review-task/comment/comment.controller.ts [hunk 2]
```diff
@@ -1,38 +1,49 @@
...
+import { AbilitiesGuard } from "../../auth/ability/abilities.guard"; 
 
 @Controller()
 export class CommentController {
     constructor(private commentService: CommentService) {}
 
     @ApiTags("comment")
     @Post("api/comment")
+    @UseGuards(AbilitiesGuard)
+    @CheckAbilities(new FactCheckerUserAbility())
```
[the remark this answers is not in the document]
Comment (LuizFNJ) [reply server/review-task/comment/comment.controller.ts:14]: question: At the moment there are 3 types of regular users, factCheckers (for reviewers too) and admin (for superAdmin too). Would it be correct for me to create a new type of user approaching factCheckers and admins in this way calling this created security layer?

[the remark this answers is not in the document]
Comment (LuizFNJ) [reply server/review-task/comment/comment.controller.ts:14]: We talked on daily and came to the conclusion that we are going to create an issue to separate the Reviewer and FactChecker users who currently occupy the same user in `abilities.decorator.ts`.

File: server/review-task/comment/comment.controller.ts [hunk 3]
```diff
@@ -1,38 +1,49 @@
...
     createReplyComment(@Param("id") id, @Body() body) {
         return this.commentService.createReplyComment(id, body);
     }
 
     @ApiTags("comment")
     @Put("api/comment/:id/delete-reply")
+    @UseGuards(AbilitiesGuard)
+    @CheckAbilities(new FactCheckerUserAbility())
```
[the remark this answers is not in the document]
Comment (LuizFNJ) [reply server/review-task/comment/comment.controller.ts:46]: We talked on daily and came to the conclusion that we are going to create an issue to separate the Reviewer and FactChecker users who currently occupy the same user in `abilities.decorator.ts`.

**Comment by LuizFNJ** [comment]
I created this issue to add the comment from @pepermao ; please review it when you can.
#1443

**Review by pepermao — APPROVED** [review]

**Pull request merged**